packages/react-deprecate-warning/src/WithDeprecatedComponent.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from "react"
+import { logComponent as log } from "./util"
+
+export interface withDeprecatedComponentProps {
+  name: string
+  warning: string
+}
+
+export const withDeprecatedComponent = <P extends object>(
+  WrappedComponent: React.ComponentType<P>,
+  { warning, name }: withDeprecatedComponentProps
+): React.ComponentType<P> =>
+  class extends React.Component<P> {
+    constructor(props) {
+      super(props)
+
+      log(name, warning)
```

Comment:
We can probably use the `name` property on the `WrappedComponent` instead of relying on it being passed in all the time. That's if your intention was for `name` to always be the same as the component name, of course.